### PR TITLE
Remove undocked solution.

### PR DESCRIPTION
**Pull request by shankarseal**
## Description

This PR consolidates two solution files into one by deleting the ebpf-for-windows-undocked.sln. The ebpf-for-windows.sln now includes the onebranch.vcxproj, which only runs if `/p:BuildOneBranch='True'` is specified. This project only builds for native configuration and copies the binary to another output directory that is needed for builds in Microsoft official build pipelines.

The post build script is modified to generate the nuget-redist package as well has handle the native-only configuration.

## Testing

CI.

## Documentation

N/A.

## Installation

N/A.

### CODE REVIEW

**Comment by Alan-Jowett** [comment]
@shankarseal I propose we get rid of the duplicate solution file. It only existed because we originally didn't want to upstream changes from the internal fork, but given that we are now keeping the two forks strictly in synch, this solution file is now redundant.

**Comment by shankarseal** [comment]
> @shankarseal I propose we get rid of the duplicate solution file. It only existed because we originally didn't want to upstream changes from the internal fork, but given that we are now keeping the two forks strictly in synch, this solution file is now redundant.

There is also a race condition as the onebranch project is running in parallel with other jobs that is it not dependent on and the xcopy is failing due to sharing violation. Trying to figure out a solution.

File: .github/workflows/cicd.yml [hunk 1]
```diff
@@ -637,7 +637,7 @@ jobs:
 
   upload_netperf_results_azure_2022:
     needs: netperf
-    if: github.event_name == 'schedule' || github.event_name == 'workflow_dispatch'
```
Comment (Alan-Jowett) [inline .github/workflows/cicd.yml:640]: Why are you reverting this?

**Review by Alan-Jowett — CHANGES_REQUESTED** [review]
Please don't revert the upload of the netperf results.

File: tools/onebranch/onebranch.vcxproj [hunk 2]
```diff
@@ -60,77 +68,23 @@
...
-      <PreprocessorDefinitions>WIN32;_DEBUG;_CONSOLE;%(PreprocessorDefinitions)</PreprocessorDefinitions>
-      <ConformanceMode>true</ConformanceMode>
-    </ClCompile>
-    <Link>
-      <SubSystem>Console</SubSystem>
-      <GenerateDebugInformation>true</GenerateDebugInformation>
-    </Link>
+  <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='NativeOnlyDebug|x64' And '$(BuildOneBranch)'=='True'">
+    <PostBuildEvent>
+      <Command>xcopy /yie $(OutDir) $(SolutionDir)build\bin\amd64chk</Command>
```
Comment (shpalani) [inline tools/onebranch/onebranch.vcxproj:74]: For my understanding, what is the purpose of amd64chk folder?

File: .github/workflows/cicd.yml [hunk 3]
```diff
@@ -637,7 +637,7 @@ jobs:
...
+    if: github.event_name == 'workflow_dispatch'
```
Comment (shpalani) [inline .github/workflows/cicd.yml:640]: upload_netperf needs to also run in 'schedule'?

File: .github/workflows/cicd.yml (hunk 1 above)
Comment (shankarseal) [reply .github/workflows/cicd.yml]: Unintentional change. Reverted.

File: tools/onebranch/onebranch.vcxproj (hunk 2 above)
Comment (shankarseal) [reply tools/onebranch/onebranch.vcxproj:74]: The microsoft internal branch requires the output goes to amd64(fre/chk) directories.

**Comment by shankarseal** [comment]
> Please don't revert the upload of the netperf results.

Reverted un-intentional change.

**Review by shpalani — APPROVED** [review]

**Review by Alan-Jowett — APPROVED** [review]

**Review by dthaler — APPROVED** [review]

**Pull request merged**